### PR TITLE
panlint: refine space after minus sign tests

### LINKED ISSUE #187: panlint: unwanted space minus operator after function
```pan
"RealMemory" = to_long(total_ram() * 0.98) - 2;
```
gives
```
Unwanted space after minus sign (not operator)
```

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #187

### CODE REVIEW

File: panc/src/main/scripts/panlint/tests.py [hunk 1]
```diff
@@ -50,13 +54,17 @@ def test_merge_diagnoses(self):
...
+        Test all files in test_files that start with test_*.pan using lint_file
+        """
         no_errors = ([], 0)
         dir_base = join(dirname(argv[0]), 'test_files')
-        self.assertEqual(panlint.lint_file(join(dir_base, 'test_good_ordinary.pan')), no_errors)
-        self.assertEqual(panlint.lint_file(join(dir_base, 'test_good_object.pan')), no_errors)
-        self.assertEqual(panlint.lint_file(join(dir_base, 'test_good_structure.pan')), no_errors)
-        self.assertEqual(panlint.lint_file(join(dir_base, 'test_good_unique.pan')), no_errors)
-        self.assertEqual(panlint.lint_file(join(dir_base, 'test_good_declaration.pan')), no_errors)
+        for afn in glob.glob(dir_base + '/test_*.pan'):
```
Comment (jrha) [inline panc/src/main/scripts/panlint/tests.py]: For cross-platform happiness this should really be:
```python
        for afn in glob.glob(join(dir_base, 'test_*.pan')):
```

File: panc/src/main/scripts/panlint/panlint.py [hunk 2]
```diff
@@ -99,9 +99,16 @@ def whitespace_around_operators(self, line, string_ranges):
 
             valid = True
             if op == '-' and \
-               re.search(r'\W\s*$', chars_before) and \
-               re.search(r'^\s*\d+\s*\W', chars_after):
+               ((re.search(r'[^\w\s)=]\s*$', chars_before) and re.search(r'^\s*\d+\s*[^\w\s]', chars_after)) or \
+                (re.search(r'=\s*$', chars_before) and re.search(r'^\s*\d+', chars_after))):
```
Comment (jrha) [inline panc/src/main/scripts/panlint/panlint.py]: Line joining is implicit within brackets, so more readable (and fewer code-climate errors) would be:
```python
if op == '-' and (
        (re.search(r'[^\w\s)=]\s*$', chars_before) and re.search(r'^\s*\d+\s*[^\w\s]', chars_after))
        or
        (re.search(r'=\s*$', chars_before) and re.search(r'^\s*\d+', chars_after))
    ):
    … code continues here …
```

Or you even go as far as:
```python
if op == '-' and (
        (
            re.search(r'[^\w\s)=]\s*$', chars_before)
            and
            re.search(r'^\s*\d+\s*[^\w\s]', chars_after)
        )
        or
        (
            re.search(r'=\s*$', chars_before)
            and
            re.search(r'^\s*\d+', chars_after)
        )
    ):
    … code continues here …
```
But maybe that's going a bit too far.

File: panc/src/main/scripts/panlint/tests.py [hunk 3]
```diff
@@ -113,9 +129,6 @@ def test_whitespace_around_operators(self):
         self.assertEqual(lc.whitespace_around_operators(bad_negative, []),
                          (False, dgn_negative, 'Unwanted space after minus sign (not operator)'))
```
Comment (jrha) [inline panc/src/main/scripts/panlint/tests.py]: Remove this blank line to fix code climate's pep8 error:
> Method definitions inside a class are separated by a single blank line.

**Comment by jrha** [comment]
Comments are advisory, code is fine, but would be shinier if codeclimate went :green_heart:.

**Comment by stdweird** [comment]
@jrha fixed

**Pull request merged**